src/coreclr/classlibnative/bcltype/arraynative.h

```diff
@@ -53,15 +51,11 @@ class ArrayNative
     };
 
     // The following functions are all helpers for ArrayCopy
-    static AssignArrayEnum CanAssignArrayType(const BASEARRAYREF pSrc, const BASEARRAYREF pDest);
-    static void CastCheckEachElement(BASEARRAYREF pSrc, unsigned int srcIndex, BASEARRAYREF pDest, unsigned int destIndex, unsigned int length);
-    static void BoxEachElement(BASEARRAYREF pSrc, unsigned int srcIndex, BASEARRAYREF pDest, unsigned int destIndex, unsigned int length);
-    static void UnBoxEachElement(BASEARRAYREF pSrc, unsigned int srcIndex, BASEARRAYREF pDest, unsigned int destIndex, unsigned int length);
-    static void PrimitiveWiden(BASEARRAYREF pSrc, unsigned int srcIndex, BASEARRAYREF pDest, unsigned int destIndex, unsigned int length);
-
+    static AssignArrayEnum CanAssignArrayType(const TypeHandle pSrc, const TypeHandle pDest);
```

Comment:
Is there a need for this outside of the translation unit?

Reply:
It seems the `private` was removed. Static private members in C++ are rarely needed unless private types are involved. In this case putting back the `private`, removing the declaration from the header, and marking the function `static` in the translation unit is preferred.

Reply:
I called it from the QCALL method for convenience. Folding it into QCALL method and replace `return` with `goto`? Declaring friend function seems unnecessary.

Reply:
There is no need to fold it into the QCall or change the QCall itself. All that needs to happen is remove this declaration and place it at the top of .cpp file where the QCall exists.

Reply:
Please also copy the `enum AssignArrayEnum` enumeration into the .cpp.

Reply:
See the suggestion in the translation unit (.cpp). This declaration and the enumeration can now be removed from this file.